### PR TITLE
Implement rustfmt example

### LINKED ISSUE #141: New example: rustfmt
Create a new example that runs the [rustfmt](https://github.com/rust-lang/rustfmt) formatter in an Oak Node, and a sample client that sends a string containing a snippet of Rust code and gets back its formatted version.

### PR DESCRIPTION
**Pull request by tiziano88**
This required fixing the following dependencies in order to compile to
wasm32:

- https://github.com/BurntSushi/ripgrep/pull/1327
- https://github.com/soc/dirs-rs/pull/24

Also requires using the nightly rust compiler because rustfmt relies on
unstable features.

Fixes #141

### CODE REVIEW

File: examples/rustfmt/module/rust/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2018 The Project Oak Authors
```
Comment (michael-kernel-sanders) [inline examples/rustfmt/module/rust/src/lib.rs]: 2019

File: Dockerfile [hunk 2]
```diff
@@ -27,10 +27,13 @@ RUN chmod a+rwx $RUSTUP_DIR
 # Install Rust toolchain.
 ARG RUST_VERSION=1.36.0
 RUN rustup toolchain install $RUST_VERSION
+# Also install the nightly toolchain as it may be needed by some examples.
+RUN rustup toolchain install nightly
```
Comment (hansman) [inline Dockerfile]: 💯

Comment (anghelcovici) [inline Dockerfile]: But.. but.. reproducible builds and all that :)

File: examples/rustfmt/client/BUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    deps = [
+        "//examples/rustfmt/proto:rustfmt_cc_grpc",
+        "//examples/utils",
+        "//oak/client:application_client",
+        "//oak/client:manager_client",
+        "@com_github_grpc_grpc//:grpc++",
+        "@com_google_absl//absl/flags:flag",
+        "@com_google_absl//absl/flags:parse",
+        "@com_google_absl//absl/memory",
+        "@com_google_asylo//asylo/util:logging",
```
Comment (anghelcovici) [inline examples/rustfmt/client/BUILD]: :( I know we don't have anything different, but having asylo as a dependency for client code is not great. I'll try to bring glog in today.

File: Dockerfile (hunk 2 above)
Comment (hansman) [reply Dockerfile]: Is there a release scheduled/foreseeable of rust compiler with the features that require nightly? 🤔

**Pull request merged**